### PR TITLE
add documentation for BitRangeExt

### PR DESCRIPTION
**Pull request by serban300**
Issue #, if available: #944 

Description of changes: add documentation for BitRangeExt

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cpuid/src/bit_helper.rs [hunk 1]
```diff
@@ -32,6 +32,8 @@ pub struct BitRange {
     pub lsb_index: u32,
 }
 
+/// Trait containing helper methods for BitRange
```
Comment (andreeaflorescu) [inline cpuid/src/bit_helper.rs]: Can you give a bit of insight into the **why** we have this Trait?

File: cpuid/src/lib.rs [hunk 2]
```diff
@@ -5,7 +5,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the THIRD-PARTY file.
 
-#![warn(missing_docs)]
+#![deny(missing_docs)]
```
Comment (andreeaflorescu) [inline cpuid/src/lib.rs]: To fix the #944 issue, can you also replace warn with deny in the other crates as well?

File: cpuid/src/bit_helper.rs (hunk 1 above)
Comment (aghecenco) [inline cpuid/src/bit_helper.rs]: nit:
```suggestion
/// Trait containing helper methods for [`BitRange`](struct.BitRange.html).
```
This should generate a hyperlink in the documentation.

File: cpuid/src/lib.rs (hunk 2 above)
Comment (serban300) [reply cpuid/src/lib.rs]: Done

File: cpuid/src/bit_helper.rs (hunk 1 above)
Comment (serban300) [reply cpuid/src/bit_helper.rs]: Done

Comment (serban300) [reply cpuid/src/bit_helper.rs]: Done

**Pull request merged**